### DISCUSSION TITLE
High Detect CPU usage

### DISCUSSION THREAD

**Discussion by jjuzwik** [blakeblackshear/frigate#15290] [Ask A Question] [answered]
Hi everyone, I have:

- Lenovo m700 i3-6100T mini-pc
- coral USB TPU
- host is latest ubuntu+docker
- recordings/clips storage is NFS share to a QNAP
- 2x Amcrest 5MP POE
- 1x Hikvision 2MP POE

I am wondering if I am just pushing the limits of my hardware. I keep getting "high CPU detect" from the GUI. 

Before I try different hardware, does anyone see anything wrong in the config or hardware setup?

[frigate config m700.txt](https://github.com/user-attachments/files/17969635/frigate.config.m700.txt)

![Screenshot 2024-12-01 at 11-58-46 Cameras Stats - Frigate](https://github.com/user-attachments/assets/0c3660d7-df68-47c3-b2c7-7f6549a27edd)
![Screenshot 2024-12-01 at 11-58-26 General Stats - Frigate](https://github.com/user-attachments/assets/7bd0f037-b83a-42a4-9942-489229788dcd)
![Screenshot 2024-12-01 at 11-58-01 General Stats - Frigate](https://github.com/user-attachments/assets/c5ca9de9-d8e9-4b94-8605-ba8a4cde6a32)

**Comment by hawkeye217** [comment]
You could try tuning your motion detection settings with the Motion Tuner for the `Front 1` camera. There seems to be constant detections happening on that camera.

**Reply by jjuzwik** [reply]
In debug view, yes, I see tons of unwanted detections happening. In the motion tuner, there are some detections but about what I would consider normal. I am using a motion mask in the area that is triggering all of the detections shown in debug view. 

It's as if 'detect' is ignoring my motion mask?

![Screenshot 2024-12-01 at 13-07-37 Object Settings - Frigate](https://github.com/user-attachments/assets/0ce74b04-a25b-46ba-b9d0-846f5a497094)
![Screenshot 2024-12-01 at 13-07-23 Motion Tuner - Frigate](https://github.com/user-attachments/assets/d1650b0d-585c-4f0b-ae47-228708183945)
![Screenshot 2024-12-01 at 13-07-01 Mask and Zone Editor - Frigate](https://github.com/user-attachments/assets/ae87987f-c36d-4876-aee9-85e36deb0f27)

**Reply by hawkeye217** [reply]
A motion mask does not stop objects from being detected. See [this tip](https://github.com/blakeblackshear/frigate/discussions/13040). 

I'd suggest setting a `min_size` for `car` objects so that small cars in that area of the frame are not tracked.

**Reply by jjuzwik** [reply]
Oh right, I think I knew that:) Umm, is this the config you are talking about:

![image](https://github.com/user-attachments/assets/c7821de7-4304-4cdf-a352-3decd6a8e33b)

**Reply by hawkeye217** [reply, marked as answer]
Yes, that's the one. Something like this:

```
objects:
  filters:
    car:
      min_area: 2000
```

**Reply by jjuzwik** [reply]
this is working great so far... I added a person filter too.  Last question, is my hardware ok in your opinion?

**Reply by hawkeye217** [reply]
Great, glad it's working for you.

You have no skipped frames and your CPU usage is generally low, so I'd say your hardware is totally fine.
